### PR TITLE
[REVIEW] Hiding implementation details for sparse API

### PR DESCRIPTION
**Pull request by cjnolet**

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

File: cpp/include/raft/sparse/convert/detail/dense.cuh [hunk 1]
```diff
@@ -83,8 +84,8 @@ void csr_to_dense(cusparseHandle_t handle, value_idx nrows, value_idx ncols,
                   cudaStream_t stream, bool row_major = true) {
   if (!row_major) {
     /**
-     * If we need col-major, use cusparse.
-     */
+                         * If we need col-major, use cusparse.
```
Comment (divyegala) [inline cpp/include/raft/sparse/convert/detail/dense.cuh:87]: This comment formatting looks off

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 2]
```diff
@@ -66,8 +66,8 @@ class COO {
   Index_Type n_cols;
 
   /**
-    * @param stream: CUDA stream to use
-    */
+              * @param stream: CUDA stream to use
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:69]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 3]
```diff
@@ -77,13 +77,13 @@ class COO {
...
 
   /**
-    * @param rows: coo rows array
-    * @param cols: coo cols array
-    * @param vals: coo vals array
-    * @param nnz: size of the rows/cols/vals arrays
-    * @param n_rows: number of rows in the dense matrix
-    * @param n_cols: number of cols in the dense matrix
-    */
+              * @param rows: coo rows array
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:80]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 4]
```diff
@@ -95,12 +95,12 @@ class COO {
       n_cols(n_cols) {}
 
   /**
-    * @param stream: CUDA stream to use
-    * @param nnz: size of the rows/cols/vals arrays
-    * @param n_rows: number of rows in the dense matrix
-    * @param n_cols: number of cols in the dense matrix
-    * @param init: initialize arrays with zeros
-    */
+              * @param stream: CUDA stream to use
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:98]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 5]
```diff
@@ -124,18 +124,18 @@ class COO {
   ~COO() {}
 
   /**
-    * @brief Size should be > 0, with the number of rows
-    * and cols in the dense matrix being > 0.
-    */
+              * @brief Size should be > 0, with the number of rows
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:127]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 6]
```diff
@@ -124,18 +124,18 @@ class COO {
...
   bool validate_size() const {
     if (this->nnz < 0 || n_rows < 0 || n_cols < 0) return false;
     return true;
   }
 
   /**
-    * @brief If the underlying arrays have not been set,
-    * return false. Otherwise true.
-    */
+              * @brief If the underlying arrays have not been set,
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:136]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 7]
```diff
@@ -146,23 +146,23 @@ class COO {
   }
 
   /*
-   * @brief Returns the rows array
-   */
+             * @brief Returns the rows array
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:149]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 8]
```diff
@@ -146,23 +146,23 @@ class COO {
...
+             */
   Index_Type *rows() { return this->rows_arr.data(); }
 
   /**
-   * @brief Returns the cols array
-   */
+             * @brief Returns the cols array
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:154]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 9]
```diff
@@ -146,23 +146,23 @@ class COO {
...
+             */
   Index_Type *cols() { return this->cols_arr.data(); }
 
   /**
-   * @brief Returns the vals array
-   */
+             * @brief Returns the vals array
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:159]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 10]
```diff
@@ -146,23 +146,23 @@ class COO {
...
+             */
   T *vals() { return this->vals_arr.data(); }
 
   /**
-    * @brief Send human-readable state information to output stream
-    */
+              * @brief Send human-readable state information to output stream
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:164]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 11]
```diff
@@ -188,53 +188,53 @@ class COO {
   }
 
   /**
-    * @brief Set the number of rows and cols
-    * @param n_rows: number of rows in the dense matrix
-    * @param n_cols: number of columns in the dense matrix
-    */
+              * @brief Set the number of rows and cols
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:191]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 12]
```diff
@@ -188,53 +188,53 @@ class COO {
...
   void setSize(int n_rows, int n_cols) {
     this->n_rows = n_rows;
     this->n_cols = n_cols;
   }
 
   /**
-    * @brief Set the number of rows and cols for a square dense matrix
-    * @param n: number of rows and cols
-    */
+              * @brief Set the number of rows and cols for a square dense matrix
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:201]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 13]
```diff
@@ -188,53 +188,53 @@ class COO {
...
     this->n_cols = n;
   }
 
   /**
-    * @brief Allocate the underlying arrays
-    * @param nnz: size of underlying row/col/val arrays
-    * @param init: should values be initialized to 0?
-    * @param stream: CUDA stream to use
-    */
+              * @brief Allocate the underlying arrays
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:210]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 14]
```diff
@@ -188,53 +188,53 @@ class COO {
...
   }
 
   /**
-    * @brief Allocate the underlying arrays
-    * @param nnz: size of the underlying row/col/val arrays
-    * @param size: the number of rows/cols in a square dense matrix
-    * @param init: should values be initialized to 0?
-    * @param stream: CUDA stream to use
-    */
+              * @brief Allocate the underlying arrays
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:220]: Formatting

File: cpp/include/raft/sparse/detail/coo.cuh [hunk 15]
```diff
@@ -188,53 +188,53 @@ class COO {
...
 
   /**
-    * @brief Allocate the underlying arrays
-    * @param nnz: size of the underlying row/col/val arrays
-    * @param n_rows: number of rows in the dense matrix
-    * @param n_cols: number of columns in the dense matrix
-    * @param init: should values be initialized to 0?
-    * @param stream: stream to use for init
-    */
+              * @brief Allocate the underlying arrays
```
Comment (divyegala) [inline cpp/include/raft/sparse/detail/coo.cuh:231]: Formatting

File: cpp/include/raft/sparse/linalg/detail/add.cuh [hunk 16]
```diff
@@ -56,9 +57,9 @@ __global__ void csr_add_calc_row_counts_kernel(
     int b_stop_idx = get_stop_idx(row, m, nnz2, b_ind);
 
     /**
-         * Union of columns within each row of A and B so that we can scan through
-         * them, adding their values together.
-         */
+                 * Union of columns within each row of A and B so that we can scan through
```
Comment (divyegala) [inline cpp/include/raft/sparse/linalg/detail/add.cuh:60]: Formatting

File: cpp/include/raft/sparse/linalg/detail/spectral.cuh [hunk 17]
```diff
@@ -47,8 +47,8 @@ void fit_embedding(const raft::handle_t &handle, int *rows, int *cols, T *vals,
   CUDA_CHECK(cudaStreamSynchronize(stream));
 
   /**
-   * Raft spectral clustering
-   */
+                     * Raft spectral clustering
```
Comment (divyegala) [inline cpp/include/raft/sparse/linalg/detail/spectral.cuh:50]: Formatting

File: cpp/include/raft/sparse/selection/detail/connect_components.cuh [hunk 18]
```diff
@@ -284,10 +295,10 @@ void min_components_by_color(raft::sparse::COO<value_t, value_idx> &coo,
                              const cub::KeyValuePair<value_idx, value_t> *kvp,
                              size_t nnz, cudaStream_t stream) {
   /**
-   * Arrays should be ordered by: colors_indptr->colors_n->kvp.value
-   * so the last element of each column in the input CSR should be
-   * the min.
-   */
+             * Arrays should be ordered by: colors_indptr->colors_n->kvp.value
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/connect_components.cuh:298]: Formatting

File: cpp/include/raft/sparse/selection/detail/connect_components.cuh [hunk 19]
```diff
@@ -334,9 +345,9 @@ void connect_components(const raft::handle_t &handle,
   value_idx n_components = get_n_components(colors.data(), n_rows, stream);
 
   /**
-   * First compute 1-nn for all colors where the color of each data point
-   * is guaranteed to be != color of its nearest neighbor.
-   */
+             * First compute 1-nn for all colors where the color of each data point
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/connect_components.cuh:348]: Formatting

File: cpp/include/raft/sparse/selection/detail/connect_components.cuh [hunk 20]
```diff
@@ -346,16 +357,16 @@ void connect_components(const raft::handle_t &handle,
               n_rows, n_cols, stream, reduction_op);
 
   /**
-   * Sort data points by color (neighbors are not sorted)
-   */
+             * Sort data points by color (neighbors are not sorted)
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/connect_components.cuh:360]: Formatting

File: cpp/include/raft/sparse/selection/detail/connect_components.cuh [hunk 21]
```diff
@@ -346,16 +357,16 @@ void connect_components(const raft::handle_t &handle,
...
+             */
   // max_color + 1 = number of connected components
   // sort nn_colors by key w/ original colors
   sort_by_color(colors.data(), nn_colors.data(), temp_inds_dists.data(),
                 src_indices.data(), n_rows, stream);
 
   /**
-   * Take the min for any duplicate colors
-   */
+             * Take the min for any duplicate colors
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/connect_components.cuh:368]: Formatting

File: cpp/include/raft/sparse/selection/detail/connect_components.cuh [hunk 22]
```diff
@@ -379,11 +390,12 @@ void connect_components(const raft::handle_t &handle,
                           temp_inds_dists.data(), n_rows, stream);
 
   /**
-   * Symmetrize resulting edge list
-   */
+             * Symmetrize resulting edge list
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/connect_components.cuh:393]: Formatting

File: cpp/include/raft/sparse/selection/detail/knn.cuh [hunk 23]
```diff
@@ -147,13 +148,13 @@ class sparse_knn_t {
 
     for (int i = 0; i < n_batches_query; i++) {
       /**
-        * Compute index batch info
-        */
+                          * Compute index batch info
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/knn.cuh:151]: Formatting

File: cpp/include/raft/sparse/selection/detail/knn.cuh [hunk 24]
```diff
@@ -147,13 +148,13 @@ class sparse_knn_t {
...
+                          */
       query_batcher.set_batch(i);
 
       /**
-        * Slice CSR to rows in batch
-        */
+                          * Slice CSR to rows in batch
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/knn.cuh:156]: Formatting

File: cpp/include/raft/sparse/selection/detail/knn.cuh [hunk 25]
```diff
@@ -192,8 +193,8 @@ class sparse_knn_t {
                                   handle.get_stream());
 
         /**
-          * Slice CSR to rows in batch
-        */
+                              * Slice CSR to rows in batch
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/knn.cuh:196]: Formatting

File: cpp/include/raft/sparse/selection/detail/knn.cuh [hunk 26]
```diff
@@ -210,8 +211,8 @@ class sparse_knn_t {
           idx_batch_indices.data(), idx_batch_data.data(), handle.get_stream());
 
         /**
-           * Compute distances
-           */
+                               * Compute distances
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/knn.cuh:214]: Formatting

File: cpp/include/raft/sparse/selection/detail/knn.cuh [hunk 27]
```diff
@@ -238,8 +239,8 @@ class sparse_knn_t {
                   handle.get_stream());
 
         /**
-         * Perform k-selection on batch & merge with other k-selections
-         */
+                             * Perform k-selection on batch & merge with other k-selections
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/knn.cuh:242]: Formatting

File: cpp/include/raft/sparse/selection/detail/knn.cuh [hunk 28]
```diff
@@ -343,8 +344,8 @@ class sparse_knn_t {
                          value_idx *query_batch_indices,
                          value_t *query_batch_data, value_t *batch_dists) {
     /**
-     * Compute distances
-     */
+                     * Compute distances
```
Comment (divyegala) [inline cpp/include/raft/sparse/selection/detail/knn.cuh:347]: Formatting

**Review by divyegala — APPROVED** [review]
PR looks great! Comments are only above formatting issues for doxygen comments on files that were moved in detail.

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**